lib/api/decorators/single.rb

```diff
@@ -51,6 +51,29 @@ def initialize(model, context = {})
                exec_context: :decorator,
                render_nil: false
 
+      def self.self_link(path, title_getter: -> (*) { represented.name })
+        link :self do
+          link_object = { href: api_v3_paths.send(path, represented.id) }
+          link_object[:title] = instance_eval(&title_getter)
+
+          link_object
+        end
+      end
```

Comment:
can't we even make a sensible default here to use the first part of the representer name if nothing is provided so that one can just write:

```self_link```